### PR TITLE
Enable test-e2e by reusing cluster-api's test-e2e

### PR DESCRIPTION
**Pull request by DanteMustCode**
- added .PHONY:
    - make test-e2e;
- redirect .PHONY:
    - make generate-e2e-templates,
    - make controller-gen,
    - make kustomize,
    - make setup-envtest,
    - make ginkgo,
    - make golangci-lint;
- updated `go.mod`:
    - upgraded `sigs.k8s.io/cluster-api/test` as `v1.7.0-rc.0` to fix a breaking change in `e2e/config/docker.yaml`,
    - upgraded `sigs.k8s.io/cluster-api` as `v1.7.0-rc.0` to catch up with `sigs.k8s.io/cluster-api/test`;
- copied `test/e2e/common.go`;
- removed duplicate files.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,6 +22,251 @@ CONTAINER_TOOL ?= docker
...
+IMPORT_BOSS_BIN := import-boss
+IMPORT_BOSS_VER := v0.28.1
+IMPORT_BOSS := $(abspath $(CAPI_TOOLS_BIN_DIR)/$(IMPORT_BOSS_BIN))
+IMPORT_BOSS_PKG := k8s.io/code-generator/cmd/import-boss
+
+CONVERSION_VERIFIER_BIN := conversion-verifier
+CONVERSION_VERIFIER := $(abspath $(CAPI_TOOLS_BIN_DIR)/$(CONVERSION_VERIFIER_BIN))
+
+OPENAPI_GEN_VER := 70dd376
+OPENAPI_GEN_BIN := openapi-gen
```
Comment (mogliang) [inline Makefile]: Can you make a cleanup on makefile, remove any unnecessary variables defintions?

File: Makefile [hunk 2]
```diff
@@ -150,66 +390,61 @@ undeploy: kustomize ## Undeploy controller from the K8s cluster specified in ~/.
...
+	    -e2e.skip-resource-cleanup=$(SKIP_RESOURCE_CLEANUP) -e2e.use-existing-cluster=$(USE_EXISTING_CLUSTER)
+
+.PHONY: $(CONTROLLER_GEN_BIN)
+$(CONTROLLER_GEN_BIN): capi-$(CONTROLLER_GEN_BIN)
+
+.PHONY: capi-$(CONTROLLER_GEN_BIN)
+capi-$(CONTROLLER_GEN_BIN):
+	$(MAKE) -C cluster-api $(CONTROLLER_GEN_BIN)
+
+.PHONY: $(KUSTOMIZE_BIN)
```
Comment (mogliang) [inline Makefile:281]: Do we need these phony? Or maybe add a few description in readme.md for how to use e2e test?

File: Makefile (hunk 1 above)
Comment (DanteMustCode) [reply Makefile]: Removed.

File: Makefile (hunk 2 above)
Comment (DanteMustCode) [reply Makefile:281]: > Do we need these phony?

They are useful if you would like to run a specific step with a specific tool. Run `make help` if you would like to discover a few tools.

> Or maybe add a few description in readme.md for how to use e2e test?

Just run `make test-e2e` and you are good to go.

**Review by mogliang — APPROVED** [review]

**Pull request merged**